### ISSUE TITLE
Can I help update this package to `zod@3`?

### ISSUE THREAD

**Issue by jensmeindertsma** [colinhacks/tozod#13]
Hello @colinhacks 👋! This package does exactly what I want. Thank you so much for putting it together. However, it looks like the demo on the README no longer works:
```ts
import { toZod } from "tozod";
import { z } from "zod";

type Player = {
  name: string;
  age?: number | undefined;
  active: boolean | null;
};

export const Player: toZod<Player> = z.object({
  name: z.string(),
  age: z.number().optional(),
  active: z.boolean().nullable(),
});
```

This fails on `tsc` with the following error:
```
Type 'ZodObject<{ name: ZodString; age: ZodOptional<ZodNumber>; active: ZodNullable<ZodBoolean>; }, "strip", ZodTypeAny, { ...; }, { ...; }>' is missing the following properties from type 'ZodObject<{ name: ZodString; age: ZodOptional<ZodNumber>; active: ZodNullable<ZodBoolean>; }, "passthrough", ZodTypeAny, Player, Player>': toJSON, primitives, nonprimitives
```

I'd love to use this package. Is there anything I can do to help update this package so that it works again?

**Comment by jensmeindertsma** [comment]
Update, I've discovered `tozod@next` which works with `zod@3`. Do you have any plans on updating this repository to reflect the latest version published to NPM?

**Comment by colinhacks** [comment]
`tozod` is dangerous and opinionated and I wouldn't recommend it. 

- At the end of the day, there are infinite ways of representing any given TS type in Zod
- It goes against the whole goal of Zod, which is to avoid duplicate static + runtime type declarations
- It's incomplete and doesn't yet include newer Zod schema types like Map, Record, etc
- TypeScript tends to change how infinitely recursive type aliases are handled between minor versions, which can cause breaking changes and obscure errors
- `tozod` doesn't (and can't) support transformations/refinements 

Basically it's terrible.

That said I just published `tozod@next` as `tozod@3`. If you want to make any enhancements here I'm happy to merge them. But consider `tozod` experimental and not officially supported.

**Comment by fern64** [comment]
Okay, thanks for elaborating.
